src/common/types/vector.cpp

```diff
@@ -554,22 +555,13 @@ Value Vector::GetValueInternal(const Vector &v_p, idx_t index_p) {
 		for (idx_t i = offlen.offset; i < offlen.offset + offlen.length; i++) {
 			children.push_back(child_vec.GetValue(i));
 		}
-		return Value::LIST(ListType::GetChildType(type), move(children));
+		return AttachTypeInfo(*vector, Value::LIST(ListType::GetChildType(type), move(children)));
 	}
 	default:
 		throw InternalException("Unimplemented type for value access");
 	}
 }
 
-Value Vector::GetValue(const Vector &v_p, idx_t index_p) {
-	auto value = GetValueInternal(v_p, index_p);
-	// set the alias of the type to the correct value, if there is a type alias
-	if (v_p.GetType().HasAlias()) {
```

Comment:
The alias inheriting already happens here, no?

Reply:
It does and it doesn't.

Since we can also have nested vectors (dictionary for example), the TypeInfo might not be part of the DictionaryVector right?

Reply:
The `GetValue` itself should generally do this - I think an assertion here would be very appropriate, however (asserting that the type is equal to the type of the vector).

Reply:
I've added the assertion and tested that it doesn't break (it did break when the typeid was AggregateState, hence the weird if statement). And I changed it to use CopyAuxInfo instead of altering/creating a new TypeInfo object